### PR TITLE
trivial: amd-gpu: read 1 byte instead of 0 bytes

### PR DESCRIPTION
**Pull request by superm1**
This should help with failures triggering the write process in the kernel.

Suggested-by: Richard Hughes <richard@hughsie.com>

Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [x] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

**Comment by hughsie** [comment]
Did it work?

**Comment by superm1** [comment]
> Did it work?

It sure did.  Look a the second commit and you'll see the crazyness I did to test it ;)

File: plugins/amd-gpu/fu-amd-gpu-device.c [hunk 1]
```diff
@@ -201,13 +201,20 @@ fu_amd_gpu_device_prepare_firmware(FuDevice *device,
...
-		return NULL;
+		} else {
+			g_set_error(error,
+				    FWUPD_ERROR,
+				    FWUPD_ERROR_NOT_SUPPORTED,
+				    "firmware for %s does not match %s",
+				    fw_pn,
+				    self->vbios_pn);
+			return NULL;
+		}
```
Comment (hughsie) [inline plugins/amd-gpu/fu-amd-gpu-device.c:217]: I think check for `if ((flags & FWUPD_INSTALL_FLAG_FORCE) == 0) {` and `return NULL` -- then unindent the g_warning.  I think your code is the wrong logic around.

Comment (superm1) [reply plugins/amd-gpu/fu-amd-gpu-device.c:217]: Oh yeah;  will fix.

**Pull request merged**